scheduler/src/cook/regexp_tools.clj

```diff
@@ -30,3 +30,16 @@
         (get field-name))
     (catch Exception e
       (throw (ex-info "Failed matching key" {:regexp-name regexp-name :field-name field-name :match-list match-list :key key} e)))))
+
+(defn match-based-on-pool-name
+  "Given a list of dictionaries [{:pool-regexp .. :field ...} {:pool-regexp .. :field ...}
+   a pool name and a <field> name, return the first matching <field> where the regexp matches the pool name."
+  [match-list effective-pool-name field & {:keys [default-value] :or {default-value nil}}]
+  (let [value (match-based-on-regexp
```

Comment:
The defaulting won't work. When nothing matches, the regexp code above throws, that distinguishes between nil result and not specified. You could pass the default up to match-based-on-regexp and have it return that.

Reply:
When nothing matches, the regexp code doesn't throw, it returns `nil`:

```
(cook.regexp-tools/match-based-on-regexp :pool-regex :field-name [{:pool-regex "^pool-name$" :field-name "foo"}] "pool-name")
=> "foo"

(cook.regexp-tools/match-based-on-regexp :pool-regex :field-name [{:pool-regex "^pool-name$" :field-name "foo"}] "non-match")
=> nil
```

Reply:
We have a pattern (prior to this PR) where `nil` is returned when no match is found, and I think that is appropriate and can remain.

Reply:
I don't think we need to support distinguishing between a configured match where the configured value is `nil` vs. no match.